aptos-move/block-executor/src/view.rs

```diff
@@ -254,12 +310,38 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TResourceVi
 
                 if matches!(mv_value, ReadResult::Uninitialized) {
                     let from_storage = self.base_view.get_state_value(state_key)?;
+                    let maybe_patched_from_storage = match (from_storage, maybe_layout) {
+                        // There are aggregators / aggregator snapshots in the
+                        // resource, so we have to replace the actual values with
+                        // identifiers.
+                        // TODO(aggregator): gate by the flag.
+                        (Some(state_value), Some(layout)) => {
+                            let res = self.replace_values_with_identifiers(state_value, layout);
+                            if let Err(err) = &res {
+                                // TODO(aggregator): This means replacement failed
```

Comment:
@igor-aptos has a nice new Error for this returned from Aggregator MVHashMap, we should re-use the flow / types for all of those.

Reply:
Do you mean we want to re-use `CommitError` here? Here if we fail (which should not happen), we always have to fallback to recovery mode